### PR TITLE
`azurerm_cosmosdb_account` - support new version of `mongo_server_version`

### LINKED ISSUE #27605: Support for CosmosDB API Mongo 7.0
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Currently, azurerm_cosmosdb_account support cosmosdb until version 4.2

Is it planned tu upgrade further version?

### New or Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_account

### Potential Terraform Configuration

```hcl
uodating attribute: mongo_server_version
```

### References

_No response_

### PR DESCRIPTION
**Pull request by sinbai**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```
PASS: TestAccCosmosDBAccount_mongoVersion70 (972.01s)
PASS: TestAccCosmosDBAccount_mongoVersion60 (949.42s)
PASS: TestAccCosmosDBAccount_mongoVersion50 (1157.10s)
```

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_cosmosdb_account` - support new version of `mongo_server_version`[GH-27605]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #27605

### CODE REVIEW

**Review by kosoniu — APPROVED** [review]

**Review by michalwilk96 — APPROVED** [review]

**Review by anton-x-t — APPROVED** [review]

**Comment by anton-x-t** [comment]
@katbyte Hi! Thank you for earlier work in this repository. Can you please review this pull request? :)

File: internal/services/cosmos/cosmosdb_account_resource_test.go [hunk 1]
```diff
@@ -3942,6 +3987,111 @@ resource "azurerm_cosmosdb_account" "test" {
...
+    consistency_level = "%s"
+  }
+
+  geo_location {
+    location          = azurerm_resource_group.test.location
+    failover_priority = 0
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, string(consistency))
+}
```
Comment (stephybun) [inline internal/services/cosmos/cosmosdb_account_resource_test.go:4093]: These test configs are all identical except for the mongo server version. Can you please parameterize this test so we can avoid the duplication and pass the mongo server version in as an argument.

**Review by stephybun — CHANGES_REQUESTED** [review]

File: internal/services/cosmos/cosmosdb_account_resource_test.go (hunk 1 above)
Comment (sinbai) [reply internal/services/cosmos/cosmosdb_account_resource_test.go]: Hi @stephybun thanks for your time and feedback. Sure, I have parameterized the test. Could you please take another look? Thanks.

**Review by stephybun — APPROVED** [review]
Thanks @sinbai LGTM 🦕

**Comment by anton-x-t** [comment]
Thank you very much @sinbai and @stephybun 👍 👍 👍

**Pull request merged**